src/test/regress/expected/citus_local_tables.out

```diff
@@ -814,6 +770,288 @@ NOTICE:  truncate cascades to table "referencing_table_xxxxxxx"
+CREATE TABLE partitioned_mx_2 PARTITION OF partitioned_mx FOR VALUES FROM (5) TO (8);
+SELECT citus_add_local_table_to_metadata('partitioned_mx');
+ citus_add_local_table_to_metadata
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT start_metadata_sync_to_node('localhost', :worker_1_port);
+ start_metadata_sync_to_node
+---------------------------------------------------------------------
+
+(1 row)
+
+CREATE TABLE partitioned_mx_3 PARTITION OF partitioned_mx FOR VALUES FROM (9) TO (12);
+\c - - - :worker_1_port
+SELECT relname FROM pg_class WHERE relname LIKE 'partitioned_mx%' ORDER BY relname;
+        relname
+---------------------------------------------------------------------
+ partitioned_mx
+ partitioned_mx_1
+ partitioned_mx_1_a_key
+ partitioned_mx_2
+ partitioned_mx_2_a_key
+ partitioned_mx_3
+ partitioned_mx_3_a_key
+ partitioned_mx_a_key
+(8 rows)
+
+\c - - - :master_port
+SET search_path TO citus_local_tables_test_schema;
+SET client_min_messages TO ERROR;
+DROP TABLE partitioned_mx;
+RESET client_min_messages;
+-- test cascading via foreign keys
+CREATE TABLE cas_1 (a INT UNIQUE);
+CREATE TABLE cas_par (a INT UNIQUE) PARTITION BY RANGE(a);
+CREATE TABLE cas_par_1 PARTITION OF cas_par FOR VALUES FROM (1) TO (4);
+CREATE TABLE cas_par_2 PARTITION OF cas_par FOR VALUES FROM (5) TO (8);
+ALTER TABLE cas_par_1 ADD CONSTRAINT fkey_cas_test_1 FOREIGN KEY (a) REFERENCES cas_1(a);
+CREATE TABLE cas_par2 (a INT UNIQUE) PARTITION BY RANGE(a);
+CREATE TABLE cas_par2_1 PARTITION OF cas_par2 FOR VALUES FROM (1) TO (4);
+CREATE TABLE cas_par2_2 PARTITION OF cas_par2 FOR VALUES FROM (5) TO (8);
+ALTER TABLE cas_par2_1 ADD CONSTRAINT fkey_cas_test_2 FOREIGN KEY (a) REFERENCES cas_1(a);
+-- this should error out as we should call the conversion from the parent
+SELECT citus_add_local_table_to_metadata('cas_par2_2');
+ERROR:  cannot convert the table because it is a partition
+-- these two should error out as the foreign keys are not inherited from the parent
+SELECT citus_add_local_table_to_metadata('cas_par2');
+ERROR:  cannot cascade operation via foreign keys as partition table citus_local_tables_test_schema.cas_par_1 involved in a foreign key relationship that is not inherited from it's parent table
+SELECT citus_add_local_table_to_metadata('cas_par2', cascade_via_foreign_keys=>true);
+ERROR:  cannot cascade operation via foreign keys as partition table citus_local_tables_test_schema.cas_par_1 involved in a foreign key relationship that is not inherited from it's parent table
+-- drop the foreign keys and establish them again using the parent table
+ALTER TABLE cas_par_1 DROP CONSTRAINT fkey_cas_test_1;
+ALTER TABLE cas_par2_1 DROP CONSTRAINT fkey_cas_test_2;
+ALTER TABLE cas_par ADD CONSTRAINT fkey_cas_test_1 FOREIGN KEY (a) REFERENCES cas_1(a);
+ALTER TABLE cas_par2 ADD CONSTRAINT fkey_cas_test_2 FOREIGN KEY (a) REFERENCES cas_1(a);
```

Comment:
We can't convert this one because the constraint is not inherited from the parent table. However, we can test something like:
`ALTER TABLE cas_par3 ADD CONSTRAINT fkey_cas_test_3 FOREIGN KEY (a) REFERENCES cas_par(a);`